### PR TITLE
adding quotaExceeded method in ApiErrorExtractor

### PR DESCRIPTION
**Pull request by danielxjd**
adding an additional method in ApiErrorExtractor to help determine if an IOException is caused by quotaExceeded. This additional method would help a lot for the development of Apache Beam.

### CODE REVIEW

File: util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java [hunk 1]
```diff
@@ -205,6 +207,18 @@ public boolean rateLimited(IOException e) {
     return false;
   }
 
+  /**
+   * Determines if a given Throwable is caused by Quota Exceeded. Recursively checks
+   * getCause() if outer exception isn't an instance of the correct class.
+   */
+  public boolean quotaExceeded(IOException e) {
+    ErrorInfo errorInfo = getErrorInfo(e);
+    if(errorInfo !=null){
```
Comment (medb) [inline util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java:216]: Please, format this change with google java formatter: https://github.com/google/google-java-format

Comment (danielxjd) [reply util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java]: Reformatted with the IntelliJ google-java-format plugin

Comment (medb) [reply util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java]: Please format only changes lines, not whole file: https://github.com/google/google-java-format/blob/master/scripts/google-java-format-diff.py

Comment (danielxjd) [reply util/src/main/java/com/google/cloud/hadoop/util/ApiErrorExtractor.java]: Sorry for that, I have created a new PR for the change, will close this one once that get passed.

**Pull request closed without merging**